packages/playground/blueprints/src/lib/steps/install-plugin.ts

```diff
@@ -58,110 +61,134 @@ export const installPlugin: StepHandler<InstallPluginStep<File>> = async (
+				pluginFolderName = file;
+				break;
+			}
 		}
 
-		/**
-		 * Pair the site editor's nested iframe to the Service Worker.
-		 *
-		 * Without the patch below, the site editor initiates network requests that
-		 * aren't routed through the service worker. That's a known browser issue:
-		 *
-		 * * https://bugs.chromium.org/p/chromium/issues/detail?id=880768
-		 * * https://bugzilla.mozilla.org/show_bug.cgi?id=1293277
-		 * * https://github.com/w3c/ServiceWorker/issues/765
-		 *
-		 * The problem with iframes using srcDoc and src="about:blank" as they
-		 * fail to inherit the root site's service worker.
-		 *
-		 * Gutenberg loads the site editor using <iframe srcDoc="<!doctype html">
-		 * to force the standards mode and not the quirks mode:
-		 *
-		 * https://github.com/WordPress/gutenberg/pull/38855
-		 *
-		 * This commit patches the site editor to achieve the same result via
-		 * <iframe src="/doctype.html"> and a doctype.html file containing just
-		 * `<!doctype html>`. This allows the iframe to inherit the service worker
-		 * and correctly load all the css, js, fonts, images, and other assets.
-		 *
-		 * Ideally this issue would be fixed directly in Gutenberg and the patch
-		 * below would be removed.
-		 *
-		 * See https://github.com/WordPress/wordpress-playground/issues/42 for more details
-		 */
-		if (
-			(await playground.isDir(
-				'/wordpress/wp-content/plugins/gutenberg'
-			)) &&
-			!(await playground.fileExists('/wordpress/.gutenberg-patched'))
-		) {
-			await playground.writeFile('/wordpress/.gutenberg-patched', '1');
-			await updateFile(
-				playground,
-				`/wordpress/wp-content/plugins/gutenberg/build/block-editor/index.js`,
-				(contents) =>
-					contents.replace(
-						/srcDoc:("[^"]+"|[^,]+)/g,
-						'src:"/wp-includes/empty.html"'
-					)
-			);
-			await updateFile(
+		if (!pluginFolderName) {
+			throw new Error('Extracted plugin folder not found');
+		}
+
+		// Move it to site plugins
+		const rootPath = await playground.documentRoot;
+		const pluginPath = `${rootPath}/wp-content/plugins/${pluginFolderName}`;
+
+		await playground.mv(tmpPluginPath, pluginPath);
+
+		const activate = 'activate' in options ? options.activate : true;
+
+		if (activate) {
+			await activatePlugin(
 				playground,
-				`/wordpress/wp-content/plugins/gutenberg/build/block-editor/index.min.js`,
-				(contents) =>
-					contents.replace(
-						/srcDoc:("[^"]+"|[^,]+)/g,
-						'src:"/wp-includes/empty.html"'
-					)
+				{
+					pluginPath,
+				},
+				progress
 			);
 		}
+
+		await maybeApplyGutenbergPatch(playground);
 	} catch (error) {
 		console.error(
-			`Proceeding without the ${pluginZipFile.name} theme. Could not install it in wp-admin. ` +
+			`Proceeding without the ${zipNiceName} plugin. Could not install it in wp-admin. ` +
 				`The original error was: ${error}`
 		);
 		console.error(error);
 	}
 };
 
+async function maybeApplyGutenbergPatch(playground: UniversalPHP) {
```

Comment:
Good call moving it to a separate function! Btw this just got fixed in Gutenberg so in a few WordPress releases we’ll be able to remove this patch:

https://github.com/WordPress/gutenberg/pull/50875#event-9328881154

Reply:
Updated the comment to reflect current status, with link to upstream PR that fixes the issue.